modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/api/classloader/ArtifactClassLoaderResolver.java

```diff
@@ -74,52 +82,43 @@ MuleDeployableArtifactClassLoader createDomainClassLoader(DomainDescriptor descr
    * @param descriptor                the descriptor of the application to generate a classLoader for.
    * @param domainClassLoaderResolver a wrapper function for {@link #createDomainClassLoader(DomainDescriptor, BiFunction)}. An
    *                                  {@link Optional#empty()} input means that the application does not use a domain.
-   * @param pluginClassLoaderResolver a wrapper function around
-   *                                  {@link #createMulePluginClassLoader(MuleDeployableArtifactClassLoader, ArtifactPluginDescriptor, Function, BiFunction)}.
-   *                                  This is usually just a call to
-   *                                  {@link #resolvePluginClassLoader(ArtifactClassLoader, ArtifactPluginDescriptor)}.
    * @return a classLoader for an application.
    */
   MuleDeployableArtifactClassLoader createApplicationClassLoader(ApplicationDescriptor descriptor,
-                                                                 Function<Optional<BundleDescriptor>, MuleDeployableArtifactClassLoader> domainClassLoaderResolver,
-                                                                 BiFunction<ArtifactClassLoader, ArtifactPluginDescriptor, ArtifactClassLoader> pluginClassLoaderResolver);
+                                                                 Function<Optional<BundleDescriptor>, ArtifactClassLoader> domainClassLoaderResolver);
```

Comment:
a Function that takes an Optional is quite weird... Shouldn't this be an `Optional<Function>` instead?